funnel/views/commentvote.py

```diff
@@ -46,19 +50,98 @@ def votedown(self):
+                    if parent.user.email:
+                        if parent.user == self.obj.user:  # check if parent comment & proposal owner are same
+                            if not current_auth.user == parent.user:  # check if parent comment is by proposal owner
+                                send_mail_info.append({'to': self.obj.user.email or self.obj.email,
+                                    'subject': u"{project} Funnel: {proposal}".format(project=self.obj.project.title, proposal=self.obj.title),
+                                    'template': 'proposal_comment_reply_email.md'})
+                        else:  # send mail to parent comment owner & proposal owner
+                            if not parent.user == current_auth.user:
+                                send_mail_info.append({'to': parent.user.email,
+                                    'subject': u"{project} Funnel: {proposal}".format(project=self.obj.project.title, proposal=self.obj.title),
+                                    'template': 'proposal_comment_to_proposer_email.md'})
+                            if not self.obj.user == current_auth.user:
+                                send_mail_info.append({'to': self.obj.user.email or self.obj.email,
+                                    'subject': u"{project} Funnel: {proposal}".format(project=self.obj.project.title, proposal=self.obj.title),
+                                    'template': 'proposal_comment_email.md'})
+
+                    if parent and parent.commentset == self.obj.commentset:
+                        comment.parent = parent
+                else:  # for top level comment
+                    if not self.obj.user == current_auth.user:
+                        send_mail_info.append({'to': self.obj.user.email or self.obj.email,
+                            'subject': u"{project} Funnel: {proposal}".format(project=self.obj.project.title, proposal=self.obj.title),
+                            'template': 'proposal_comment_email.md'})
+                self.obj.commentset.count += 1
+                comment.voteset.vote(current_auth.user)  # Vote for your own comment
+                db.session.add(comment)
+                flash(_("Your comment has been posted"), 'info')
+            db.session.commit()
+            to_redirect = self.obj.url_for(_external=True)
+            for item in send_mail_info:
+                email_body = render_template(item.pop('template'), proposal=self.obj, comment=comment, link=to_redirect)
+                if item.get('to'):
+                    # Sender is set to None to prevent revealing email.
+                    send_mail(sender=None, body=email_body, **item)
+        # Redirect despite this being the same page because HTTP 303 is required to not break
+        # the browser Back button
+        return redirect(to_redirect, code=303)
+
+    @route('comments/delete', methods=['POST'])
+    @lastuser.requires_login
+    @requires_permission('new_comment')
```

Comment:
Wrong permission.

Reply:
`delete_comment` is not available here for the reason posted above. The right permission is being checked inside the view for same reason. The comments edit/delete code in the front end needs to change to use right permission.

Reply:
I assume this is fixed now? Didn't notice it when reading through the code.

Reply:
Yeah. This is fixed.